src/postings/block_search.rs

```diff
@@ -213,29 +68,41 @@ mod tests {
             targets.insert(val);
         }
         for target in targets {
-            util_test_search_in_block(block_searcher, block, target);
+            util_test_search_in_block(block, target);
         }
     }
 
     fn search_in_block_trivial_but_slow(block: &[u32], target: u32) -> usize {
         block.iter().take_while(|&&val| val < target).count()
     }
 
-    fn test_search_in_block_util(block_searcher: BlockSearcher) {
+    #[test]
+    fn test_search_in_branchless_binary_search() {
         for len in 1u32..128u32 {
             let v: Vec<u32> = (0..len).map(|i| i * 2).collect();
-            util_test_search_in_block_all(block_searcher, &v[..]);
+            util_test_search_in_block_all(&v[..]);
         }
     }
 
     #[test]
-    fn test_search_in_block_scalar() {
-        test_search_in_block_util(BlockSearcher::Scalar);
-    }
-
-    #[cfg(target_arch = "x86_64")]
-    #[test]
-    fn test_search_in_block_sse2() {
-        test_search_in_block_util(BlockSearcher::Sse2);
+    fn test_search_specific_array() {
+        let block: [u32; COMPRESSION_BLOCK_SIZE] = [
+            5, 23, 41, 48, 52, 93, 103, 135, 157, 158, 166, 169, 171, 189, 226, 228, 239, 244, 250,
+            252, 265, 267, 272, 290, 320, 323, 336, 338, 356, 361, 368, 375, 411, 412, 416, 422,
+            425, 428, 430, 460, 462, 464, 466, 476, 490, 492, 508, 511, 552, 558, 588, 590, 601,
+            623, 635, 636, 661, 671, 681, 683, 684, 689, 706, 707, 725, 730, 744, 748, 753, 765,
+            770, 772, 773, 778, 782, 787, 791, 798, 802, 804, 811, 817, 834, 844, 852, 855, 861,
+            862, 863, 885, 888, 891, 913, 914, 960, 964, 967, 970, 986, 992, 1027, 1032, 1034,
+            1041, 1048, 1061, 1062, 1069, 1072, 1079, 1082, 1083, 1102, 1140, 1165, 1170, 1176,
+            1189, 1195, 1198, 1200, 1201, 1208, 1223, 1240, 1252, 1276, 1307,
+        ];
+        for target in block.iter().flat_map(|&el| vec![el - 1, el].into_iter()) {
```

Comment:
I would add 
`vec![el - 1, el, el+1]` to test with a jump size larger than 1 to the next bucket (except the last element .take(block.len() * 3 -1) )

Reply:
el +1 is the same as el-1 for the precedent element.

Reply:
I'll test all elements in [0..1308).

Reply:
> Simpler and faster, really cool. One minor comment. Also I don't know if the data is monotonically increasing or if we can have duplicates.

Good point I modified the contract. There was an error in it too.